packages/react/src/components/DateTimePicker/DateTimePicker.story.jsx

```diff
@@ -46,193 +42,156 @@ export default {
-        }}
-        hasTimeInput={boolean('hasTimeInput', true)}
-        showRelativeOption={false}
-        onApply={action('onApply')}
-        onCancel={action('onCancel')}
-      />
-    </div>
+      hasTimeInput={boolean('hasTimeInput', true)}
+      showRelativeOption={false}
+      onApply={action('onApply')}
+      onCancel={action('onCancel')}
+      hasIconOnly={boolean('hasIconOnly', false)}
+    />
   );
 };
 
 WithoutARelativeOption.storyName = 'Without a relative option';
 
 export const WithoutACustomRangeLink = () => {
-  const size = select('size', Object.keys(CARD_SIZES), CARD_SIZES.MEDIUMWIDE);
   return (
-    <div
-      style={{
-        width: `${getCardMinSize('lg', size).x}px`,
-        margin: spacing06,
+    <DateTimePicker
+      id="datetimepicker"
+      defaultValue={{
+        timeRangeKind: PICKER_KINDS.PRESET,
+        timeRangeValue: PRESET_VALUES[3],
       }}
-    >
-      <DateTimePicker
-        id="datetimepicker"
-        defaultValue={{
-          timeRangeKind: PICKER_KINDS.PRESET,
-          timeRangeValue: PRESET_VALUES[3],
-        }}
-        hasTimeInput={boolean('hasTimeInput', true)}
-        showCustomRangeLink={false}
-        onApply={action('onApply')}
-        onCancel={action('onCancel')}
-      />
-    </div>
+      hasTimeInput={boolean('hasTimeInput', true)}
+      showCustomRangeLink={false}
+      onApply={action('onApply')}
+      onCancel={action('onCancel')}
+      hasIconOnly={boolean('hasIconOnly', false)}
+    />
   );
 };
 
 WithoutACustomRangeLink.storyName = 'Without a custom range link';
 
 export const LightVersion = () => {
-  const size = select('size', Object.keys(CARD_SIZES), CARD_SIZES.MEDIUMWIDE);
   return (
-    <div
-      style={{
-        width: `${getCardMinSize('lg', size).x}px`,
-        margin: spacing06,
-      }}
-    >
-      <DateTimePicker
-        id="datetimepicker"
-        dateTimeMask={text('dateTimeMask', 'YYYY-MM-DD HH:mm')}
-        relatives={[
-          {
-            label: 'Yesterday',
-            value: RELATIVE_VALUES.YESTERDAY,
-          },
-        ]}
-        light
-        hasTimeInput={boolean('hasTimeInput', true)}
-      />
-    </div>
+    <DateTimePicker
+      id="datetimepicker"
+      dateTimeMask={text('dateTimeMask', 'YYYY-MM-DD HH:mm')}
+      relatives={[
+        {
+          label: 'Yesterday',
+          value: RELATIVE_VALUES.YESTERDAY,
+        },
+      ]}
+      light
+      hasTimeInput={boolean('hasTimeInput', true)}
+      hasIconOnly={boolean('hasIconOnly', false)}
+    />
   );
 };
 
 LightVersion.storyName = 'Light version';
 
+export const IconOnly = () => {
+  return (
+    <DateTimePicker
+      id="datetimepicker"
+      dateTimeMask={text('dateTimeMask', 'YYYY-MM-DD HH:mm')}
+      relatives={[
+        {
+          label: 'Yesterday',
+          value: RELATIVE_VALUES.YESTERDAY,
+        },
+      ]}
+      hasTimeInput={boolean('hasTimeInput', true)}
+      hasIconOnly={boolean('hasIconOnly', true)}
```

Comment:
Can we add the onApply/onCancel storybook action here? It's slightly confusing when setting an absolute time range to click "Apply" and the window immediately close. With the iconOnly version we don't have that visual feedback to see the correct range was selected. I can confirm it does by turning `hasIconOnly` off for a second to see the range, but still...

Reply:
Added